src/overlays/actors/ovl_Bg_Lotus/z_bg_lotus.c

```diff
@@ -19,22 +28,151 @@ const ActorInit Bg_Lotus_InitVars = {
     (ActorFunc)BgLotus_Init,
     (ActorFunc)BgLotus_Destroy,
     (ActorFunc)BgLotus_Update,
-    (ActorFunc)BgLotus_Draw
+    (ActorFunc)BgLotus_Draw,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/ovl_Bg_Lotus_0x80AD6760/BgLotus_Init.asm")
+extern BgMeshHeader D_06000A20; // Lilypad collision
+extern Gfx D_06000040[];        // Lilypad model
+
+void BgLotus_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgLotus* this = THIS;
+    s32 pad;
+    s32 sp2C;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    BcCheck3_BgActorInit(&this->dyna, 1);
+    BgCheck3_LoadMesh(globalCtx, &this->dyna, &D_06000A20);
+    this->dyna.actor.floorHeight =
+        func_800C411C(&globalCtx->colCtx, &thisx->floorPoly, &sp2C, &this->dyna.actor, &this->dyna.actor.world.pos);
+    this->timer2 = 96;
+    this->dyna.actor.world.rot.y = (s16)(rand() >> 0x10);
+    this->actionFunc = BgLotus_Wait;
+}
+
+void BgLotus_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgLotus* this = THIS;
+
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void BgLotus_SetScaleXZ(BgLotus* this) {
+    f32 scale;
+
+    if (this->dyna.actor.params == 0) {
+        scale = sin_rad(this->timer * 0.7853982f) * ((0.014f * ((f32)this->timer / 8)) + 0.01f);
```

Comment:
Does still match if you do `scale = sin_rad(this->timer * 0.7853982f) * ((0.014f * (this->timer / 8.0f)) + 0.01f);`? (no cast)